src/backend/src/services/projects.services.ts

```diff
@@ -15,8 +15,30 @@ import {
 } from '../utils/projects.utils';
 import { descBulletConverter, wbsNumOf } from '../utils/utils';
 import { createDescriptionBulletChangesJson } from '../utils/work-packages.utils';
+import WorkPackagesService from './work-packages.services';
 
 export default class ProjectsService {
+  /**
+   * Delete the the project in the database along with all its work packages.
+   * @returns the project that is deleted.
+   */
+  static async deleteProject(user: User, _wbsNumber: WbsNumber): Promise<WbsNumber> {
+    if (!isProject(_wbsNumber)) throw new HttpException(400, `${wbsPipe(_wbsNumber)} is not a valid project WBS #!`);
+    if (user.role === Role.GUEST || user.role === Role.LEADERSHIP || user.role === Role.MEMBER) {
+      throw new AccessDeniedException('Guests, Members, and Leadership cannot delete projects');
+    }
+
+    const deletedProject = await prisma.wBS_Element.delete({
```

Comment:
i understand the approach you're getting at but prisma.project.update does not have dateDeleted or deletedByUser, so i ended up going around it a little. i used prisma.project.find... to ensure the project exists and to extract its project id (for later uses), but then i used prisma.wbs_element.update so that i can assign the dateDeleted and deletedBy.

let me know if this is similar to what u were saying